docs/proposals/registry/registry-operator.md

```diff
@@ -0,0 +1,101 @@
+# Devfile Registry Operator Proposal
+
+## Problem
+Currently, the devfile registry is deployed onto a running cluster by `kubectl apply`-ing a few yaml files. This works fine for a proof of concept or doing a quick demo, but there’s a few problems with this approach that make it infeasible long-term:
+
+1) Configuration is difficult.
+   
+   For example, the current approach makes it difficult to programatically configure TLS and authentication for the OCI registry server when deploying.
+
+2) Deployment isn't user friendly and hard to tell how/why something went wrong
+
+3) Updates to the registry instance are not straightfoward
+
+    Requires the registry administrator to manually edit and update the devfile registry deployment resource.
+
+3) Deployment isn’t the same on Kubernetes and OpenShift.
+
+   To handle this, we currently have to maintain two separate sets of yaml files for deploying the registry on Kubernetes or OpenShift.
+
+## Solution
+Implementing an operator to programatically deploy devfile registries will streamline the deployment process on both Kubernetes and OpenShift, as well as provide greater flexibility for customization and configuration when deploying it (especially as we add support for deploying with TLS and authentication enabled). It will also provide us the ability to onboard the devfile registry to OperatorHub in the future, if we wish.
+
+## DevfileRegistry Custom Resource
+
+### API Group:
+`registry.devfile.io`
+
+### API Version
+`v1alpha1`
+
+### Resource Kind
+`DevfileRegistry`
+
+### Scope
+`Namespaced`
+
+### Fields
+
+#### Registry 
```

Comment:
I assume the `Registry` here is an additional field that includes the two images? IIRC that is one of our old fields design and currently we move the two images one level up without `Registry` field

Reply:
`Registry` here doesn't have any binding on the format of the custom resource spec, it's just a header to group fields together.

Reply:
But actually, since the `registry` header is kind of ambiguous and confusing, I'll remove it and move the two fields up one level to match the spec.

Reply:
Okay, from the header level perspective, I saw `Registry`, `Storage` and `TLS` are under header level `####` and the header level `Fields` is `###` so I interpret it to the format below:
```yaml
spec:
  registry:
    bootstrapImage: ''
    ociRegistryImage: ''
  tls:
    enabled: true
    secretName: ''
  storage:
    enabled: true
    registryVolumeSize: '3Gi'
```

But that's fine if we just use it for group the definitions